### PR TITLE
Topic recognition - Theo Duval (44792701) DCGAN on OASIS Dataset

### PR DESCRIPTION
**Pull request by tduval840**
Hi Shakes,

I have written an implementation of a deep convolutional generative adversarial network to produce images based on the preprocessed OASIS dataset. Included are two files generator.py and discriminator.py that contain the models I used, as well as a file train.py to handle loading the data, training the model and producing/saving results. Please see the README.md file for more information on some of the implementation decisions I made, as well as usage, system requirements and some general information on the task I've solved. Feel free to provide any feedback on what I've done!

Thanks,
Theo

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages 
- Code check out for Hard Difficulty 
- Results are good

Other Feedback
- Good grouping of files and resources 
- Good use of Keras custom model object  
- Tip you dont need 2 nested gradient tapes, you can use the same tape but with a keyword `persistent=True` when building it

**Comment by rexsw** [comment]
Code looks good for hard difficulty
Good description on your pull request too

Comments like:
   def call(self, x):
        """Define a single call of this model."""
Don’t add to the reliability of your code and should be left out. However, there are lots of good comments then do help improve the readability of the code as well.

Tensorflow has many image processing functions that you can use rather than using an additional librariesies like PIL and numpy

**Comment by shakes76** [comment]
## TF Usage
Good
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Good work
